File: ts-solutions/tests/1493_longest_subarray.test.ts
import { longestSubarray } from '../src/1493_longest_subarray'

type LongestSubarrayTest = {
	nums: number[]
	want: number
}

const tests: LongestSubarrayTest[] = [
	{ nums: [1, 1, 0, 1], want: 3 },
	{ nums: [0, 1, 1, 1, 0, 1, 1, 0, 1], want: 5 },
	{ nums: [1, 1, 1], want: 2 },
]

describe(`longestSubarray`, () => {
	test.each(tests)(`longestSubarray($nums) => $want`, ({ nums, want }) => {
		expect(longestSubarray(nums)).toEqual(want)
	})
})

File: go-solutions/1493_longest_subarray_test.go
package leetcode

import "testing"

type longestSubarrayTest struct {
	nums []int
	want int
}

var tests = []longestSubarrayTest{
	{[]int{1, 1, 0, 1}, 3},
	{[]int{0, 1, 0, 1}, 2},
	{[]int{0, 1, 1, 1, 0, 1, 1, 0, 1}, 5},
	{[]int{1, 1, 1}, 2},
}

func TestLongestSubarray(t *testing.T) {
	for _, subarrayTest := range tests {
		if got := LongestSubarray(subarrayTest.nums); got != subarrayTest.want {
			t.Errorf("got %d, want %d, passed %v", got, subarrayTest.want, subarrayTest.nums)
		}
	}
}
